astropy/coordinates/representation.py

```diff
@@ -1579,14 +1579,20 @@ def from_cartesian(cls, cart):
     def represent_as(self, other_class, differential_class=None):
         # Take a short cut if the other class is a spherical representation
 
-        # TODO: this could be optimized to shortcut even if a differential_class
-        # is passed in, using the ._re_represent_differentials() method
+        # TODO: implement here after #11469
         if inspect.isclass(other_class) and not differential_class:
             if issubclass(other_class, PhysicsSphericalRepresentation):
-                return other_class(phi=self.lon, theta=90 * u.deg - self.lat, r=1.0,
-                                   copy=False)
+                # diffs = self._re_represent_differentials(other_class,
```

Comment:
I think part of the reason this fails is that `baseframe` asks for something impossible - to always transform to `SphericalDifferential`. Probably best to raise a separate issue...

Reply:
So can we merge before this is resolved, and then followup PR to uncomment those lines, or just wait?

Reply:
See #11482

Reply:
I suspect https://github.com/astropy/astropy/pull/11467#issuecomment-813764255 will not be trivial. Perhaps the best route is to not use ``_re_represent_differentials`` for this function as it preserves the unit keys.

Reply:
Not quite sure how much it really helps, but FWIW, tests seem to pass if I write the if-statement as:
```
            elif (issubclass(other_class, SphericalRepresentation)
                  and (not differential_class
                       or all(diff.base_representation is self.__class__
                              for diff in self.differentials.values()))):
```
That at least is closer to what we want...

Reply:
Interesting 🤔 ... Obviously it's up to you and the maintainers, but my personal preference would be not not complicate the if-blocks, merge in this PR without optimizing the UnitSpherical case, and then tackle the problem after that... But happy any which way 👍 .

Reply:
I agree my suggestion is not great either, so I'm OK with simply not addressing UnitSpherical here. But in that case I think we should remove the commented out code - since that does not work right now, I don't think it is helpful. A more detailed comment about why it is tricky (maybe just referring to this discussion) would also be good.

Reply:
I just removed the commented out code in representations.py, but left it in test_representations.py as that code can just be uncommented as soon as we find a UnitsSpherical fix.